### DISCUSSION TITLE
Evaluating DeepSeek chat model

### DISCUSSION THREAD

**Discussion by KaoutharBousbaa1** [community/community#181252] [Programming Help]
### Discussion Type

Question

### Discussion Content

Hi!

I have a dataset with expected outputs, and I need to evaluate the DeepSeek-Chat model to see whether it labels the outputs correctly. Unlike OpenAI Evals, I couldn’t find any built-in evaluation tools for DeepSeek. Could you please advise if there is a way to run evaluations, or how to best approach this?

Thank you so much!

**Comment by zeelkher** [comment]
1. Prepare Your Dataset

Create a file (CSV or JSON) with two columns:

input – the prompt or question

expected_output – the correct answer

Example:

input | expected_output
Classify: cat | animal

2. Run DeepSeek on Each Input

Send each input to the DeepSeek-Chat model and save what the model returns.

Now you will have three fields:

input

expected_output

model_output

3. Compare the Answers

Check whether:

model_output == expected_output

If yes → correct
If no → incorrect

4. Calculate Accuracy

Use this simple formula:

accuracy = correct_answers / total_samples

Example:
80 correct out of 100 → 80% accuracy

**Comment by mecodeatlas** [comment]
Thanks for posting in the GitHub Community, @KaoutharBousbaa1!

We're happy you're here. You are more likely to get a useful response if you are posting your question in the applicable category, the Discussions category is solely related to conversations around the [GitHub product Discussions](https://docs.github.com/en/discussions/collaborating-with-your-community-using-discussions/about-discussions). This question should be in the `Programming Help` category. I've gone ahead and moved it for you. Good luck!

**Comment by TechnicalDeveloper** [comment]
Hi!

You’re right – DeepSeek doesn’t (yet) ship a built-in eval system like OpenAI Evals, but you can still evaluate it in the usual ML way since you already have a dataset with expected outputs.

The general approach:

Decide what you’re measuring

If the model is supposed to output a label (class name, “yes/no”, etc.), use classic metrics:
accuracy, precision/recall, F1, confusion matrix.

If the output is free text, you can either:

map it back to a label (e.g. via simple rules / regex), or

use another model / a rubric as a “judge” to score correctness.

Wrap the DeepSeek-Chat model call

Write a small script (Python, JS, etc.) that:

Reads your dataset: input + expected_output

Calls the DeepSeek API / model for each input

Normalizes the model output (trim whitespace, lowercase, maybe map synonyms)

Compares it to the expected output

For labeling tasks, set temperature = 0 so the model is as deterministic as possible.

Compute metrics over the dataset
Pseudocode in Python for a simple classification case:

import csv

def call_deepseek(prompt: str) -> str:
    # TODO: replace with your actual DeepSeek-Chat call
    # return model_output_text
    ...

gold = []
preds = []

with open("dataset.csv") as f:
    reader = csv.DictReader(f)
    for row in reader:
        prompt = row["input"]
        expected = row["label"]

        raw_output = call_deepseek(prompt)
        pred = raw_output.strip().lower()
        gold_label = expected.strip().lower()

        gold.append(gold_label)
        preds.append(pred)

# simple accuracy
correct = sum(p == g for p, g in zip(preds, gold))
accuracy = correct / len(gold)
print(f"Accuracy: {accuracy:.3f}")


You can then plug gold and preds into any standard metrics library
(e.g. sklearn.metrics or evaluate) to get precision/recall/F1, etc.

(Optional) Use an existing eval framework
If you prefer something more structured, you can:

Use an open-source eval harness (e.g. lm-eval-harness or a custom eval framework),

Implement a small “DeepSeekModel” adapter that knows how to call DeepSeek,

Register your dataset and run the harness as usual.

In short: even though there’s no “DeepSeek Evals” product, you can treat DeepSeek-Chat like any other model — run it over your labeled dataset, normalize its answers, and compute the usual metrics. The main work is just wiring up the API and deciding on a clear matching rule between model outputs and your gold labels.

Hope that helps!
